### PR TITLE
Admin Dashboard Refactor

### PR DESCRIPTION
**Pull request by thekabistro**

### CODE REVIEW

File: public/css/dashboard.css [hunk 1]
```diff
@@ -392,3 +392,13 @@ body {
...
+.form-check-label {
+    color: #fff;
+}
+
+.white {
+    color: #fff;
+}
+  
+  
+
```
Comment (TresTres) [inline public/css/dashboard.css]: thanks

File: src/action_creators/ActionTypes.js [hunk 2]
```diff
@@ -10,6 +10,11 @@ export const VIEW_CONTROL = {
 };
 
 
+//Admin actions
+export const ADMIN_DATA = {
+  UPDATE_DATA: 'UPDATE_DATA',
+};
+
```
Comment (TresTres) [inline src/action_creators/ActionTypes.js]: good

File: src/action_creators/AdminActions.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if(datap.statusCode === 200)
+          dispatch({
+            type: ADMIN_DATA.UPDATE_DATA,
+            data: datap.body
+          });
+        else
+          dispatch({
+            type: ADMIN_DATA.UPDATE_DATA,
+            data: 'Fatal Error'
+          });
```
Comment (TresTres) [inline src/action_creators/AdminActions.js]: This is good for now, in the future, prevent using data field for both error messages and valid data.

The store can have several fields to update.

File: src/reducers/AdminManager.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      return {
+        ...state, 
+        queryData: action.data
+      };
+    default: 
+      return state;
+  }
+};
+
+export default AdminManager;
```
Comment (TresTres) [inline src/reducers/AdminManager.js]: good

File: src/reducers/rootReducer.js [hunk 5]
```diff
@@ -3,12 +3,14 @@ import { combineReducers } from 'redux';
...
 
 const rootReducer = combineReducers({
   loginManager: LoginManager,
   userManager: UserManager,
-  viewController: ViewController
+  viewController: ViewController,
+  adminManager: AdminManager
 });
 
 export default rootReducer;
```
Comment (TresTres) [inline src/reducers/rootReducer.js]: good

File: src/smart_components/AdminDashboard.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return bindActionCreators({
+    updateData: adminActions.queryDB,
+  }, dispatch);
+}
+
+export default connect(mapStateToProps, mapDispatchToProps) (AdminDashboard);
```
Comment (TresTres) [inline src/smart_components/AdminDashboard.js]: This looks good in general, but I don't see anything regarding `VotingModal`, `ModalError`, or `AdminEmailing`.  
Apologize if that wasn't clear but we need those still - even if these aren't refactored yet, these still need to be included and rendered with the rest of the Dashboard.  You can also look at #17 for an update to `AdminEmailing`, the Dashboard is the interface for directors to admit people into attendance, etc.

Please admit this into higher priority.

Comment (TresTres) [reply src/smart_components/AdminDashboard.js]: You can reference the original `Admin.js` for this

File: src/smart_components/UserForm.js [hunk 7]
```diff
@@ -41,6 +42,7 @@ class UserForm extends React.Component{
           </div>
         </div>
         <InfoPrompt userStatus={status}/>
```
Comment (TresTres) [inline src/smart_components/UserForm.js]: Admins are users yes, but there will be no use case for them to use the `UserForm`, it doesn't need to be there at all.  They are implicitly admitted to HackRU and don't need to provide travel info.  I would suggest putting the `AdminDashboard` as a component to be separately rendered from the `UserForm` entirely based on admin status.  

If this is what the render process looks like now:
crudeflowbefore

We would like to change it to this:
crudeflowafter

**Comment by ryanisaacg** [comment]
This should be good to merge now I think

**Pull request merged**